containers/Connector/Connector.tsx

```diff
@@ -48,88 +59,52 @@ const useConnector = () => {
-    []
-  );
-
-  // Optimism Mainnet
-  const L2DefaultProvider: SynthetixProvider = useMemo(
-    () =>
-      loadProvider({
-        infuraId: process.env.NEXT_PUBLIC_INFURA_PROJECT_ID
-          ? process.env.NEXT_PUBLIC_INFURA_PROJECT_ID
-          : '0',
-        networkId: NetworkIdByName['mainnet-ovm'],
-      }),
-    []
-  );
-
-  const updateState = useCallback(
-    (update: AppState) => {
-      if (update.wallets.length > 0) {
-        const wallet = update.wallets[0].accounts[0];
+  const updateState = useCallback((update: AppState) => {
+    if (update.wallets.length > 0) {
+      const wallet = update.wallets[0].accounts[0];
 
-        const { label } = update.wallets[0];
-        const { id } = update.wallets[0].chains[0];
-        const networkId = getNetworkIdFromHex(id);
+      const { label } = update.wallets[0];
+      const { id } = update.wallets[0].chains[0];
+      const networkId = getNetworkIdFromHex(id);
 
-        const isSupported = isSupportedNetworkId(networkId) && isSupportedWalletChain(networkId);
-
-        const network = {
-          id: networkId,
-          name: isSupportedNetworkId(networkId)
-            ? NetworkNameById[networkId]
-            : 'Unsupported Network',
-          useOvm: getIsOVM(networkId),
-        };
-
-        const provider = new ethers.providers.Web3Provider(update.wallets[0].provider, {
-          name: network.name,
-          chainId: networkId,
-        });
-
-        const signer = provider.getSigner();
-        const contracts = setSynthetix(networkId, signer);
-        const synthetixjs = isSupported ? ({ contracts } as SynthetixJS) : null;
-
-        dispatch({
-          type: AppEvents.CONFIG_UPDATE,
-          payload: {
-            address: wallet.address,
-            walletWatched: null,
-            walletType: label,
-            network,
-            provider,
-            signer,
-            synthetixjs,
-            ensName: wallet?.ens?.name || null,
-            ensAvatar: wallet?.ens?.avatar?.url || null,
-          },
-        });
-
-        const connectedWallets = update.wallets.map(({ label }) => label);
-        localStorage.setItem(LOCAL_STORAGE_KEYS.SELECTED_WALLET, JSON.stringify(connectedWallets));
-      } else {
-        dispatch({ type: AppEvents.WALLET_DISCONNECTED });
-      }
-    },
-    [setSynthetix]
-  );
+      const network = {
+        id: networkId,
+        name: isSupportedNetworkId(networkId) ? NetworkNameById[networkId] : 'Unsupported Network',
+        useOvm: getIsOVM(networkId),
+      };
 
-  const transactionNotifier = useMemo(
-    () => new TransactionNotifier(L1DefaultProvider),
-    [L1DefaultProvider]
-  );
+      const provider = new ethers.providers.Web3Provider(update.wallets[0].provider, {
+        name: network.name,
+        chainId: networkId,
+      });
+
+      const signer = provider.getSigner();
+      const contracts = isSupportedNetworkId(networkId)
+        ? initializeSynthetix(networkId, signer)
```

Comment:
Minor gripe but this will be re-referenced every render

Reply:
Hmm yea but it is in a useCallback? So only when the state of onboard changes. 
I think that's only when account is changes -> new signer or when network changes. In both these cases we want "synthetix"/ the contracts to be re-initialised?

Reply:
Ah yep - my bad - updateState is already wrapper :)